test/async-emitter.js

```diff
@@ -0,0 +1,214 @@
+'use strict';
+
+const { AsyncEmitter } = require('..');
+const metatests = require('metatests');
+
+metatests.test('AsyncEmitter on/emit', async test => {
+  const ae = new AsyncEmitter();
+  let emitted = false;
+
+  ae.on('e1', async (a, b, c, d) => {
```

Comment:
```suggestion
  ae.on('e1', test.mustCall(async (a, b, c, d) => {
  ...
```

also, no need for `emitted` and `count` here and in other cases when using `mustCall`/`mustNotCall`.